src/libraries/System.Numerics.Tensors/ref/System.Numerics.Tensors.netcore.cs

```diff
@@ -157,4 +416,127 @@ public static partial class TensorPrimitives
         public static void Xor<T>(System.ReadOnlySpan<T> x, System.ReadOnlySpan<T> y, System.Span<T> destination) where T : System.Numerics.IBitwiseOperators<T, T, T> { }
         public static void Xor<T>(System.ReadOnlySpan<T> x, T y, System.Span<T> destination) where T : System.Numerics.IBitwiseOperators<T, T, T> { }
     }
+    public static partial class TensorSpanExtensions
+    {
+        public static System.Numerics.Tensors.TensorSpan<T> AsTensorSpan<T>(this T[]? array, scoped System.ReadOnlySpan<nint> shape) { throw null; }
+        public static bool SequenceEqual<T>(this System.Numerics.Tensors.TensorSpan<T> span, System.Numerics.Tensors.TensorSpan<T> other) where T : System.IEquatable<T>? { throw null; }
+    }
+    public readonly ref partial struct TensorSpan<T>
+    {
+        private readonly object _dummy;
+        private readonly int _dummyPrimitive;
+        public TensorSpan(System.Span<T> span) { throw null; }
+        public TensorSpan(System.Span<T> span, scoped System.ReadOnlySpan<nint> lengths, scoped System.ReadOnlySpan<nint> strides) { throw null; }
+        [System.CLSCompliantAttribute(false)]
+        public unsafe TensorSpan(T* data, nint dataLength) { throw null; }
+        [System.CLSCompliantAttribute(false)]
+        public unsafe TensorSpan(T* data, nint dataLength, scoped System.ReadOnlySpan<nint> lengths, scoped System.ReadOnlySpan<nint> strides) { throw null; }
+        public TensorSpan(T[]? array) { throw null; }
+        public TensorSpan(T[]? array, System.Index startIndex, scoped System.ReadOnlySpan<nint> lengths, scoped System.ReadOnlySpan<nint> strides) { throw null; }
+        public TensorSpan(T[]? array, int start, scoped System.ReadOnlySpan<nint> lengths, scoped System.ReadOnlySpan<nint> strides) { throw null; }
+        public static System.Numerics.Tensors.TensorSpan<T> Empty { get { throw null; } }
+        public nint FlattenedLength { get { throw null; } }
+        public bool IsEmpty { get { throw null; } }
+        public ref T this[scoped System.ReadOnlySpan<System.Buffers.NIndex> indexes] { get { throw null; } }
+        public System.Numerics.Tensors.TensorSpan<T> this[scoped System.ReadOnlySpan<System.Buffers.NRange> ranges] { get { throw null; } set { } }
+        public ref T this[scoped System.ReadOnlySpan<nint> indexes] { get { throw null; } }
+        [System.Diagnostics.CodeAnalysis.UnscopedRefAttribute]
+        public System.ReadOnlySpan<nint> Lengths { get { throw null; } }
+        public int Rank { get { throw null; } }
+        [System.Diagnostics.CodeAnalysis.UnscopedRefAttribute]
+        public System.ReadOnlySpan<nint> Strides { get { throw null; } }
+        public void Clear() { }
+        public void CopyTo(System.Numerics.Tensors.TensorSpan<T> destination) { }
```

Comment:
This and other span inputs should be scoped, to help indicate we won't capture it